api/apps/api/src/modules/access-control/projects-acl/project-acl.controller.ts

```diff
@@ -43,17 +47,19 @@ export class ProjectAclController {
   async findUsersInProject(
     @Param('projectId', ParseUUIDPipe) projectId: string,
     @Req() req: RequestWithAuthenticatedUser,
-  ): Promise<UserRoleInProjectDto[] | boolean> {
+    @Query('q') nameSearch?: string,
+  ): Promise<UsersInProjectResult> {
```

Comment:
this comment is not actually about a change in this PR, as I hadn't noticed this earlier, but we are not returning a _real_ JSON:API response payload here - I assume frontends are fine with this since the implementation so far wasn't going through the JSON:API serializer in any case, but I suggest to try avoiding using different response formats across the API (except for things such as GeoJSON or other non-JSON-document-kind-of content types)

Reply:
I started reading about the JSON:API payloads and examples, so there would be a way, prior to returning the actual payload, where some serializer would structure the data in the way JSON:API specifies, right? Or am I understanding this incorrectly?

Reply:
I say so because if it is used elsewhere, I can try to use or adapt it here so it follows the same principles.

Reply:
> so there would be a way, prior to returning the actual payload, where some serializer would structure the data in the way JSON:API specifies, right?

Yes, that's right - all the CRUD endpoints (or most of them, if I got distracted and didn't notice) should use JSON:API as response format (and more or less so also for GET query params).

Since I won't be around to help you with this over the coming days, and since this endpoint was not using a JSON:API response format so far in any case, I suggest to merge this PR as is, and in January I can onboard you to the JSON:API strategy we've used elsewhere so that you can apply the same in other endpoints when you get to these.

In the meanwhile if you want to have a look around, take for example `api/apps/api/src/modules/users/users.service.ts`

In most of these classes, the `get serializerConfig()` getter returns the configuration for the shape of JSON:API responses that we then feed to the serializer (`jsonapi-serializer`) through the `AppBaseService.serialize()` method.

To close the loop, have a look at `UsersController.findAll()`, for example, where the fetching of data is followed by its serialization, whose results we finally send in the response.